x-pack/plugins/security/server/lib/authorization/check_privileges_dynamically.ts

```diff
@@ -29,10 +29,15 @@ export function checkPrivilegesDynamicallyWithRequestFactory(
 ): CheckPrivilegesDynamicallyWithRequest {
   return function checkPrivilegesDynamicallyWithRequest(request: Record<string, any>) {
     const checkPrivileges = checkPrivilegesWithRequest(request);
-    return async function checkPrivilegesDynamically(privilegeOrPrivileges: string | string[]) {
+    return async function checkPrivilegesDynamically(
```

Comment:
The fact that we're using `checkPrivilegesDynamically` and passing in the `spaceId` makes me uneasy. If I understand correctly, we're doing so because when spaces is disabled a `null`/`undefined` `spaceId` requires that we authorize the user globally. Where-as when spaces is enabled, a `null`/`undefined` `spaceId` requires us to authorize the user at the default space.